code/__HELPERS/visual_effects.dm

```diff
@@ -16,7 +16,10 @@
  * This is just so you can apply the animation to things which can be animated but are not movables (like images)
  */
 #define STOP_FLOATING_ANIM(target) \
-	animate(target, transform = matrix(), time = 1 SECONDS)
+	var/datum/decompose_matrix/decomposed_target_matrix = target.transform.decompose();\
+	var/matrix/original_matrix = matrix().Turn(decomposed_target_matrix.rotation);\
+	original_matrix = original_matrix.Scale(decomposed_target_matrix.scale_x, decomposed_target_matrix.scale_y);\
+	animate(target, transform = original_matrix, time = 1 SECONDS);\
```

Comment:
```suggestion
	do { \
		var/datum/decompose_matrix/_decomposed_target_matrix = target.transform.decompose();\
		var/matrix/_original_matrix = matrix().Turn(_decomposed_target_matrix.rotation);\
		_original_matrix = _original_matrix.Scale(_decomposed_target_matrix.scale_x, _decomposed_target_matrix.scale_y);\
		animate(target, transform = _original_matrix, time = 1 SECONDS);\
	} while(FALSE)
```
do while avoids corrupting the namespace too badly, tho we do need to name the vars right to avoid potential double declarations
that or just make a proc 4head